packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -855,18 +840,10 @@ function resetChildExpirationTime(
     while (child !== null) {
       const childUpdateExpirationTime = child.expirationTime;
       const childChildExpirationTime = child.childExpirationTime;
-      if (
-        newChildExpirationTime === NoWork ||
-        (childUpdateExpirationTime !== NoWork &&
-          childUpdateExpirationTime < newChildExpirationTime)
-      ) {
+      if (childUpdateExpirationTime > newChildExpirationTime) {
         newChildExpirationTime = childUpdateExpirationTime;
       }
-      if (
-        newChildExpirationTime === NoWork ||
-        (childChildExpirationTime !== NoWork &&
-          childChildExpirationTime < newChildExpirationTime)
-      ) {
+      if (childChildExpirationTime > newChildExpirationTime) {
         newChildExpirationTime = childChildExpirationTime;
       }
```

Comment:
At this point we could even go so far as to do

```
newChildExpirationTime = Math.max(
  newChildExpirationTime,
  child.expirationTime,
  child.childExpirationTime,
);
```

so simple… wow

![doge meme](https://user-images.githubusercontent.com/6820/47263948-87aef680-d4c0-11e8-8139-cbdf1d1811fd.png)

Reply:
😢